djstripe/models.py

```diff
@@ -509,40 +509,41 @@ def sync_current_subscription(self, cu=None):
 
     def update_plan_quantity(self, quantity, charge_immediately=False):
         self.subscribe(
-            plan=djstripe_settings.plan_from_stripe_id(
-                self.stripe_customer.subscription.plan.id
-            ),
+            plan=self.stripe_customer.subscription.plan.id,
             quantity=quantity,
             charge_immediately=charge_immediately
         )
 
-    def subscribe(self, plan, quantity=1, trial_days=None,
+    def subscribe(self, stripe_plan_id, quantity=1, trial_days=None,
                   charge_immediately=True, prorate=djstripe_settings.PRORATION_POLICY):
         cu = self.stripe_customer
         """
         Trial_days corresponds to the value specified by the selected plan
         for the key trial_period_days.
         """
-        if ("trial_period_days" in djstripe_settings.PAYMENTS_PLANS[plan]):
-            trial_days = djstripe_settings.PAYMENTS_PLANS[plan]["trial_period_days"]
+
+        plan_object = Plan.objects.get(stripe_id=stripe_plan_id)
+
+        if plan_object.trial_period_days:
+            trial_days = plan_object.trial_period_days
 
         if trial_days:
             resp = cu.update_subscription(
-                plan=djstripe_settings.PAYMENTS_PLANS[plan]["stripe_plan_id"],
+                plan=plan_object.stripe_id,
                 trial_end=timezone.now() + datetime.timedelta(days=trial_days),
                 prorate=prorate,
                 quantity=quantity
             )
         else:
             resp = cu.update_subscription(
-                plan=djstripe_settings.PAYMENTS_PLANS[plan]["stripe_plan_id"],
+                plan=plan_object.stripe_id,
                 prorate=prorate,
                 quantity=quantity
             )
         self.sync_current_subscription()
         if charge_immediately:
             self.send_invoice()
-        subscription_made.send(sender=self, plan=plan, stripe_response=resp)
+        subscription_made.send(sender=self, plan=stripe_plan_id, stripe_response=resp)
```

Comment:
in this case, transmitted stripe plan id. I thought it would be logical to rename the variable, as originally I was confused and did not know what is stored in the plan parameter.